src/macaron/slsa_analyzer/package_registry/npm_registry.py

```diff
@@ -187,10 +191,20 @@ def download_attestation_payload(self, url: str, download_path: str) -> bool:
 
             logger.debug("Found attestation with valid predicateType: %s", att.get("predicateType"))
 
+            download_path_extended = download_path
+            if att.get("predicateType") == "https://github.com/npm/attestation/tree/main/specs/publish/v0.1":
+                download_path_extended = download_path_extended + ".signed"
+            else:
+                if downloaded:
+                    # In case there are multiple unsigned provenance, log it and continue.
+                    logger.debug("Found multiple unsigned provenance.")
+                    continue
+                downloaded = True
```

Comment:
Hmm the `downloaded` variable is only set to True if we encounter the first unsigned provenance. Can we rename it to something that's less generic to prevent confusion with the signed provenance? I am thinking of something like `downloaded_unsigned`.

Reply:
In additions, do we assume that there is always at least an unsigned provenance if a signed provenance is available?
For example, if `res_obj.get("attestations")` only contains the signed provenance predicate, then the return value of this function will be `False` even though we managed to download a signed provenance :thinking:

Reply:
I expect that both kinds of provenance should exist. If only the signed is found, there is nothing we can do with it and reporting `False` makes the most sense. I will add a message to the log though.